java/src/jmri/managers/DefaultShutDownManager.java

```diff
@@ -303,6 +307,10 @@ private void doShutdown(int status, boolean exit) {
                     return;
                 }
             }
+
+
+            closeFrames(start);
```

Comment:
This needs to be after the permission check below. The user might abort the shutdown during that check and therefore the frames should not be closed until after that check.